docs/common/message-workflows.md

```diff
@@ -0,0 +1,314 @@
+answer the request and wait for another response.
+
+## Initialize/Features
+
+As first message, the computer should send an empty Initialize packet
+and expect a Features packet as response. The Initialize packet will
+cause the device to stop what it is currently doing and should work at
+any time. Thus, it can also be used to recover from previous errors.
+
+## Button meta-workflow
+
+If the device requires the user to press a button, it will reply with a
+ButtonRequest to the computer. The computer should immediately send a
+ButtonAck acknowledging the request. But it should also display an
+indication to the user that it should follow the instruction on the
+device. The field code in the ButtonRequest message explains what type
+of request the user should acknowledge with a button.
+
+If the user never presses the button, there will never be a reply to the
+ButtonAck message. The computer can use Cancel to abort the current
+operation. This should result in a Failure response.
+
+## PinMatrix meta-workflow
+
+If the device requires the user to unlock the device with a PIN, it will
+reply with a PinMatrixRequest. The field type gives some explanation
+what PIN is required (current pin, new pin, or confirmation of new pin).
+The computer should display an empty pin matrix and let the user enter
+the pin. The computer should encode the PIN as if the numbers are
+ordered like they are on the numeric keypad. The encoded PIN should then
+be send with a PinMatrixAck message.
+
+## Passphrase meta-workflow
+
+If the device requires the user to enter the passphrase, it will reply
+with a PassphraseRequest. The computer should ask the user for the
+passphrase and send it in clear text with a PassphraseAck message.
+
+## GetAddress
+
+The message GetAddress (send from the computer to the device) serves two
+purposes. It can be used to get a valid address or to display the
+address on the device. The field address_n gives the bip-32
+path to the
+address. The field coin_name should be set to some
+supported coin (see the Feature message for a list of supported coins).
+For multisig addresses multisig must be filled out with all
+participating master public keys and their bip-32 path. The script_type
+field has the same meaning as for transaction inputs when signing:
+
+-   SPENDADDRESS (standard P2PKH address)
+-   SPENDMULTISIG (P2SH multisig address)
+-   SPENDWITNESS (native segwit P2WPKH or
+    multisig p2wsh address)
+-   SPENDP2SHWITNESS (segwit encapsulated in a p2sh
+    address)
+
+If show_display is set the address is displayed to the user. In any
+case, it is also sent to the computer with an Address response.
+
+## GetPublicKey
+
+The message GetPublicKey can be used to get a bip-32 master public key
+from the trezor or to display it to the user. The field address_n gives
+the bip-32 path to the master key. The field ecdsa_curve_name can be
+used to get Ed25519 or
+NIST256P1 public keys.
+
+## SignTx
```

Comment:
please add a header pointing to `communication/bitcoin-signing.md`, saying that the below might be imprecise / obsolete

Reply:
Resolved in 0268958.